packages/core/src/routes/experience/classes/verifications/code-verification.ts

```diff
@@ -209,4 +176,110 @@ export class CodeVerification
       verified,
     };
   }
+
+  abstract toUserProfile(): T extends VerificationType.EmailVerificationCode
+    ? { primaryEmail: string }
+    : { primaryPhone: string };
```

Comment:
use a type map for the return type for better extensibility

Reply:
updated